### DISCUSSION TITLE
MULTIPLE QUESTION BOT based on pollbot.py

### DISCUSSION THREAD

**Discussion by uniru** [python-telegram-bot/python-telegram-bot#2896] [Q&A]
Hello,
I am new using python-telegram-bot I am following the code from the example **pollbot.py,** in order to create a poll bot. **I would like to cycle through a list of questions (and their related lists of answers), so that the user, after answering the n-th question, will receive automatically the (n+1)-th question up to the latest question of the list.**

I managed to send all questions at once, but this is not the goal. **I need that the bot will send just one question and only follows it up with the next when it gets the answer from the user.** 
Maybe someone can help me to clarify how to solve this issue with python-telegram-bot? 

Thanks a lot in advance :)
Anna

**Comment by Poolitzer** [comment]
hey, you still want to use native polls, just with one answer each? then you need to store your questions + answers somewhere. Generate the first much like they are generated in pollbot.py poll(), the important part is that is_anonymous is False and that you store the poll_id somewhere. Then you can use receive_poll to check which answer the poll id (you match that to the one you store in the previous step) got, store that information and send the next question, again store the poll id, rinse and repeat.

**Reply by Bibo-Joshi** [reply]
dependen on how elaborate the setup is, `ConversationHandler` might be useful as well

**Reply by Poolitzer** [reply]
gotta remember to only set [per_user to True](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#what-do-the-per_-settings-in-conversationhandler-do)

**Reply by uniru** [reply]
Hello! Thanks for the feedback I will try to implement your advises and I let you know how it goes!
